File: src/main/scala/net/minecraft/world/WorldLens.scala
package net.minecraft.world

import net.minecraft.world.chunk.IChunkProvider
import net.minecraft.world.storage.loot.LootTableManager

object WorldLens {
  def isChunkLoaded(world: World, x: Int, z: Int, allowEmpty: Boolean): Boolean = world.isChunkLoaded(x, z, allowEmpty)

  def setChunkProvider(world: World, chunkProvider: IChunkProvider): Unit = {
    world.chunkProvider = chunkProvider
  }

  def setLootTable(world: World, lootTable: LootTableManager): Unit = {
    world.lootTable = lootTable
  }

}

File: src/main/scala/hohserg/dimensional/layers/lens/WorldLens.java
package hohserg.dimensional.layers.lens;

import gloomyfolken.hooklib.api.*;
import net.minecraft.world.*;
import net.minecraft.world.chunk.*;
import net.minecraft.world.storage.*;
import net.minecraft.world.storage.loot.*;
import org.apache.commons.lang3.*;

@HookContainer
public class WorldLens {

    @FieldLens
    public static FieldAccessor<World, IChunkProvider> chunkProvider;

    @FieldLens
    public static FieldAccessor<World, LootTableManager> lootTable;

    @FieldLens
    public static FieldAccessor<World, WorldInfo> worldInfo;
    
    @MethodLens
    public static boolean isChunkLoaded(World self, int var1, int var2, boolean var3){
        throw new NotImplementedException("hooklib failed to method lens");
    }
}
